atc/db/worker_factory.go

```diff
@@ -321,7 +321,20 @@ func (f *workerFactory) SaveWorker(atcWorker atc.Worker, ttl time.Duration) (Wor
 	return savedWorker, nil
 }
 
-func (f *workerFactory) FindWorkerForContainerByOwner(owner ContainerOwner) (Worker, bool, error) {
+func (f *workerFactory) FindWorkerForContainerByOwner(owner ContainerOwner, teamID int) (Worker, bool, error) {
```

Comment:
There seems to be a problem with this - in testing with a team worker, I noticed that a new container would be created every time I executed a check against the team's resources.

Here's my `docker-compose.yml` for setting up a team worker:

```yaml
version: '3'

services:
  db:
    image: postgres
    ports: ["6543:5432"]
    environment:
    - POSTGRES_DB=concourse
    - POSTGRES_USER=dev
    - POSTGRES_PASSWORD=

  web:
    build:
      context: .
      dockerfile: ${DOCKERFILE}
    image: concourse/concourse:local
    volumes:
    - .:/src
    command: web
    depends_on: [db]
    ports: ["8080:8080"]
    environment:
    - CONCOURSE_POSTGRES_HOST=db
    - CONCOURSE_POSTGRES_USER=dev
    - CONCOURSE_POSTGRES_PASSWORD=dev
    - CONCOURSE_POSTGRES_DATABASE=concourse
    - CONCOURSE_EXTERNAL_URL=http://localhost:8080
    - CONCOURSE_ADD_LOCAL_USER=test:test,guest:guest
    - CONCOURSE_MAIN_TEAM_LOCAL_USER=test
    - CONCOURSE_LOG_LEVEL=debug
    - CONCOURSE_ENABLE_GLOBAL_RESOURCES=true

  worker:
    build:
      context: .
      dockerfile: ${DOCKERFILE}
    image: concourse/concourse:local
    command: worker
    privileged: true
    depends_on: [web]
    ports: ["7777:7777", "7788:7788"]
    environment:
    - CONCOURSE_LOG_LEVEL=debug
    - CONCOURSE_TSA_HOST=web:2222
    - CONCOURSE_EPHEMERAL=${CONCOURSE_EPHEMERAL:-true}

      # avoid using loopbacks
    - CONCOURSE_BAGGAGECLAIM_DRIVER=overlay

      # so we can reach Garden/Baggageclaim for debugging
    - CONCOURSE_BIND_IP=0.0.0.0
    - CONCOURSE_BAGGAGECLAIM_BIND_IP=0.0.0.0

      # so that CI can configure a different network range that doesn't overlap
      # with the outer Garden
    - CONCOURSE_GARDEN_NETWORK_POOL

  team_worker:
    build:
      context: .
      dockerfile: ${DOCKERFILE}
    image: concourse/concourse:local
    command: worker
    privileged: true
    depends_on: [web]
    environment:
    - CONCOURSE_TEAM=some-team

    - CONCOURSE_LOG_LEVEL=debug
    - CONCOURSE_TSA_HOST=web:2222
    - CONCOURSE_EPHEMERAL=${CONCOURSE_EPHEMERAL:-true}

      # avoid using loopbacks
    - CONCOURSE_BAGGAGECLAIM_DRIVER=overlay

      # so we can reach Garden/Baggageclaim for debugging
    - CONCOURSE_BIND_IP=0.0.0.0
    - CONCOURSE_BAGGAGECLAIM_BIND_IP=0.0.0.0

      # so that CI can configure a different network range that doesn't overlap
      # with the outer Garden
    - CONCOURSE_GARDEN_NETWORK_POOL
```

Then configure `some-team`:

```sh
$ fly -t l login -c http://localhost:8080 -u test -p test
$ fly -t l set-team -n some-team --local-user guest
```

Log in and set the following pipeline:

```yaml
groups: []
resources:
- name: booklit
  type: git
  source:
    uri: https://github.com/vito/booklit
- name: 3s
  type: time
  source: {interval: 3s}
resource_types: []
jobs:
- name: unit
  plan:
  - get: booklit
    trigger: true
  - get: 3s
  - task: test
    file: booklit/ci/test.yml
```

```sh
$ fly -t lst login -c http://localhost:8080 -u guest -p guest
$ fly -t lst set-pipeline -p booklit -c booklit.yml
```

Then trigger a few checks from the web UI.

There should only be two `check` containers - one for the `time` resource and one for the `git` resource. I saw one new container be created every time I ran the `check`. This can be verified by looking at the database:

```sh
$ ./hack/db
```

```sql
select rccs.id, rccs.expires_at - now() as expires_in, c.handle, c.worker_name, t.name as team_name from containers c, resource_config_check_sessions rccs, workers w left join teams t on w.team_id = t.id where w.name = c.worker_name and c.resource_config_check_session_id = rccs.id;
```

Side note: in my local testing I also configured the same pipeline under the `main` team. Not sure if that matters.